flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/runner/JobDispatcherLeaderProcessFactoryFactory.java

```diff
@@ -77,7 +79,15 @@ public JobDispatcherLeaderProcessFactory createFactory(
         }
 
         final JobResultStore jobResultStore = jobPersistenceComponentFactory.createJobResultStore();
-        final Collection<JobResult> recoveredDirtyJobResults = getDirtyJobResults(jobResultStore);
+
+        if (accessJobResultStore(
+                () -> jobResultStore.hasCleanJobResultEntry(jobGraph.getJobID()))) {
+            throw new FlinkRuntimeException(
+                    DuplicateJobSubmissionException.ofGloballyTerminatedAndCleaned(
```

Comment:
Why do we check this in 2 places (here and the ApplicationDispatcherBootstrap)?

It seems like we keep spreading these checks around; Dispatcher#submitJob is alkso checking for duplicate jobs...
Not sure whether this is maintainable.